src/io/flutter/utils/StdoutJsonParser.java

```diff
@@ -32,11 +33,25 @@ public void appendOutput(String string) {
       if (eatNextEol) {
         eatNextEol = false;
 
-        if (c == '\n' || c == '\r') {
+        if (c == '\n') {
+          continue;
+        }
+
+        if (c == '\r' && !isPotentialWindowsReturn) {
+          eatNextEol = true;
+          isPotentialWindowsReturn = true;
           continue;
         }
       }
 
+      if (isPotentialWindowsReturn) {
+        isPotentialWindowsReturn = false;
+
+        if (c != '\n') {
```

Comment:
a little odd that we don't flush when the `c` is a `'\n'`
is this really
`assert(c!= '\n');`
?